wandb/data_types.py

```diff
@@ -1027,25 +1035,32 @@ class Html(BatchableMedia):
             to False the HTML will pass through unchanged.
     """
 
+    artifact_type = "html-file"
+
     def __init__(self, data, inject=True):
         super(Html, self).__init__()
 
-        if isinstance(data, str):
-            self.html = data
-        elif hasattr(data, "read"):
-            if hasattr(data, "seek"):
-                data.seek(0)
-            self.html = data.read()
+        if isinstance(data, str) and os.path.exists(data):
+            self._set_file(data, is_tmp=False)
+            with open(data, "r") as file:
+                self.html = file.read()
         else:
-            raise ValueError("data must be a string or an io object")
-        if inject:
-            self.inject_head()
+            if isinstance(data, str):
+                self.html = data
+            elif hasattr(data, "read"):
+                if hasattr(data, "seek"):
+                    data.seek(0)
+                self.html = data.read()
+            else:
+                raise ValueError("data must be a string or an io object")
+            if inject:
+                self.inject_head()
```

Comment:
I think this should be moved one block higher so it's also applied to the file path case.

Reply:
Got it... I changed this to be the case in the next commit. If injecting against a local file, it will read in the data and create a new tmp file, else use the local file directly. For artifact deserialization, we explicitly pass `inject=False`.